crates/runestick/src/modules/result.rs

```diff
@@ -38,18 +40,33 @@ fn expect_impl(result: Result<Value, Value>, message: &str) -> Result<Value, VmE
 }
 
 fn and_then_impl(
-    this: &Result<Value, Value>,
+    this: Result<Value, Value>,
     then: Function,
 ) -> Result<Result<Value, Value>, VmError> {
     match this {
         Ok(v) => Ok(then.call::<_, _>((v,))?),
-        Err(e) => Ok(Err(e.clone())),
+        Err(e) => Ok(Err(e)),
     }
 }
 
-fn map_impl(this: &Result<Value, Value>, then: Function) -> Result<Result<Value, Value>, VmError> {
+fn map_impl(this: Result<Value, Value>, then: Function) -> Result<Result<Value, Value>, VmError> {
```

Comment:
Just to clarify,this means that `.map` will now take ownership of the Result type it maps over.